lib/graphql/schema/printer.rb

```diff
@@ -0,0 +1,131 @@
+module GraphQL
+  # Used to convert your {GraphQL::Schema} to a GraphQL schema string
+  #
+  # @example print your schema to standard output
+  #   Schema = GraphQL::Schema.new(query: QueryType)
+  #   puts GraphQL::Schema::Printer.print_schema(Schema)
+  #
+  module Schema::Printer
+    extend self
```

Comment:
Is this to support the two class methods, `print_schema` and `print_introspection_schema`? 

What if those were implemented as:

```ruby
def self.print_schema 
  printer =  self.new(schema, :is_defined_type)
  printer.to_s 
end

def self.print_introspection_schema 
  printer =  self.new(schema, :is_introspection_type)
  printer.to_s 
end

# and, to go with that:
def to_s
  print_filtered_schema(@schema, @type_filter)
end
```

I think the advantage there is that most methods are instance-only, rather than instance methods _and_ class methods.